pkg/scale/comparison_test.go

```diff
@@ -16,160 +16,161 @@
-		},
-		&types.SealDigest{
-			ConsensusEngineID: types.BabeEngineID,
-			Data:              []byte{1, 3, 5, 7},
-		},
-	}
-	oldEncode, err := oldDigests.Encode()
-	if err != nil {
-		t.Errorf("unexpected err: %v", err)
-		return
-	}
-
-	vdt, err := NewVaryingDataType(ChangesTrieRootDigest{}, PreRuntimeDigest{}, ConsensusDigest{}, SealDigest{})
-	if err != nil {
-		t.Errorf("unexpected err: %v", err)
-		return
-	}
-	err = vdt.Set(ChangesTrieRootDigest{
-		Hash: common.Hash{0, 91, 50, 25, 214, 94, 119, 36, 71, 216, 33, 152, 85, 184, 34, 120, 61, 161, 164, 223, 76, 53, 40, 246, 76, 38, 235, 204, 43, 31, 179, 28},
-	})
-	if err != nil {
-		t.Errorf("unexpected err: %v", err)
-		return
-	}
-
-	newDigest := []VaryingDataType{
-		mustNewVaryingDataTypeAndSet(
-			ChangesTrieRootDigest{
-				Hash: common.Hash{0, 91, 50, 25, 214, 94, 119, 36, 71, 216, 33, 152, 85, 184, 34, 120, 61, 161, 164, 223, 76, 53, 40, 246, 76, 38, 235, 204, 43, 31, 179, 28},
-			},
-			ChangesTrieRootDigest{}, PreRuntimeDigest{}, ConsensusDigest{}, SealDigest{},
-		),
-		mustNewVaryingDataTypeAndSet(
-			PreRuntimeDigest{
-				ConsensusEngineID: types.BabeEngineID,
-				Data:              []byte{1, 3, 5, 7},
-			},
-			ChangesTrieRootDigest{}, PreRuntimeDigest{}, ConsensusDigest{}, SealDigest{},
-		),
-		mustNewVaryingDataTypeAndSet(
-			ConsensusDigest{
-				ConsensusEngineID: types.BabeEngineID,
-				Data:              []byte{1, 3, 5, 7},
-			},
-			ChangesTrieRootDigest{}, PreRuntimeDigest{}, ConsensusDigest{}, SealDigest{},
-		),
-		mustNewVaryingDataTypeAndSet(
-			SealDigest{
-				ConsensusEngineID: types.BabeEngineID,
-				Data:              []byte{1, 3, 5, 7},
-			},
-			ChangesTrieRootDigest{}, PreRuntimeDigest{}, ConsensusDigest{}, SealDigest{},
-		),
-	}
-
-	newEncode, err := Marshal(newDigest)
-	if err != nil {
-		t.Errorf("unexpected err: %v", err)
-		return
-	}
-	if !reflect.DeepEqual(oldEncode, newEncode) {
-		t.Errorf("encodeState.encodeStruct() = %v, want %v", oldEncode, newEncode)
-	}
-
-	decoded := NewVaryingDataTypeSlice(vdt)
-	err = Unmarshal(newEncode, &decoded)
-	if err != nil {
-		t.Errorf("unexpected err: %v", err)
-	}
-	// decoded.Types
-	if !reflect.DeepEqual(decoded.Types, newDigest) {
-		t.Errorf("Unmarshal() = %v, want %v", decoded.Types, newDigest)
-	}
-}
-
-func BenchmarkUnmarshal(b *testing.B) {
-	for i := 0; i < b.N; i++ {
-		for _, tt := range allTests {
-			dst := reflect.New(reflect.TypeOf(tt.in)).Elem().Interface()
-			if err := Unmarshal(tt.want, &dst); (err != nil) != tt.wantErr {
-				b.Errorf("decodeState.unmarshal() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-		}
-	}
-}
-
-func BenchmarkMarshal(b *testing.B) {
-	for i := 0; i < b.N; i++ {
-		for _, tt := range allTests {
-			// dst := reflect.New(reflect.TypeOf(tt.in)).Elem().Interface()
-			if _, err := Marshal(tt.in); (err != nil) != tt.wantErr {
-				b.Errorf("decodeState.unmarshal() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-		}
-	}
-}
+//
```

Comment:
why is this commented out?

Reply:
@timwu20 Do you have suggestions on how to remove circular dependency this creates?

Reply:
You can remove this file for now